architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md

```diff
@@ -0,0 +1,74 @@
+                                |                  |     |                 |         |
+                                +---------+--------+     +-----------+-----+         |
+                                          |                          |               |
+                                          |                          |               |
+                +---------------+         |    +---------------+     |               |
+                |               |     Read|    |               |     |Read/Update    |
+                |               |         |    |               |     |               |
+                |               |         |    |               |     |               |
+                |    Publish    |         +---->  Events Bus   <-----+               |
+                |               |              |  Controller   |                     |Create/Delete
+                |               |              |               |                     |
+                +--------+------+              +--+-------+----+                     |
+                         |                        |       |Create/Delete             |
+                         |Publish   Create/Delete |       |                          |
+                         |                        |       +-----+                    |
+                     +---v------------------------v----+        |             +------+-----+
+                     |                                 |        |             |            |
+                     |                                 +--------v-------------+  Subscriber|
+                     |         KN-Channel              |    KN-Subscription   |            |
+                     |                                 +----------------------+            |
+                     |                                 |                      +------------+
+              +------+---------------------------------+--------+
+              |                                                 |
+              |                                                 |
+              |                 Knative Bus                     |
+              |                                                 |
+              +-------------------------------------------------+
+
+### NATS Streaming
+
+Since knative provides an abstraction of "The Bus" which is a pluggable component of knative eventing that have different  implementations backed up by a certain messaging store "e.g. Kafka Bus, GCP PubSub Bus, etc..".
+
+### Decision
+
+1. Implementing a production ready `NATS Streaming` knative bus implementation.
+2. Provide `NATS Streaming` operator that provides `NATS Streaming` clusters which satisfies all production needs.
+3. Implementing an `Azure Service Bus Messaging` knative bus implementation as a plan B.
+4. Contribute both implementations to `knative eventing`.
+
+## Status
+
+In Progress
+
+## Consequences
+
```

Comment:
Add an introducing sentence before the list:
`These are the consequences of this architecture decision:`